integration-tests/helpers/fake-agent.js

```diff
@@ -174,15 +174,13 @@ module.exports = class FakeAgent extends EventEmitter {
       msgCount += 1
       try {
         fn(msg)
-        if (msgCount === expectedMessageCount) {
+        if (resolveAtFirstSuccess || msgCount === expectedMessageCount) {
```

Comment:
you are changing the previous behaviour here.
if `fn(msg)` throws an Error, we are not going to remove the listener if `msgCount === expectedMessageCount`

Reply:
@iunanua Adding `resolveAtFirstSuccess` did not fix the telemetry test issue, so I reverted the changes and improved the tests to ensure that telemetry was tested on the `appsec` namespace